crates/solvers/src/tests/dex.rs

```diff
@@ -0,0 +1,113 @@
+use std::{
+    net::SocketAddr,
+    sync::{Arc, Mutex},
+};
+
+#[derive(Debug, Clone)]
+pub enum Expectation {
+    Get {
+        path: String,
+        res: serde_json::Value,
+    },
+    Post {
+        path: String,
+        req: serde_json::Value,
+        res: serde_json::Value,
+    },
+}
+
+/// Set up an mock external DEX or DEX aggregator API.
+pub async fn setup(expectations: Vec<Expectation>) -> SocketAddr {
+    let state = Arc::new(Mutex::new(expectations));
+    let app = axum::Router::new()
+        .route(
+            "/*path",
+            axum::routing::get(
+                |axum::extract::State(state),
+                 axum::extract::Path(path),
+                 axum::extract::RawQuery(query)| async move {
+                    axum::response::Json(get(state, Some(path), query))
+                },
+            )
+            .post(
+                |axum::extract::State(state),
+                 axum::extract::Path(path),
+                 axum::extract::RawQuery(query),
+                 axum::extract::Json(req)| async move {
+                    axum::response::Json(post(state, Some(path), query, req))
+                },
+            ),
+        )
+        // Annoying, but `axum` doesn't seem to match `/` with the above route,
+        // so explicitely mount `/`.
+        .route(
+            "/",
+            axum::routing::get(
+                |axum::extract::State(state), axum::extract::RawQuery(query)| async move {
+                    axum::response::Json(get(state, None, query))
+                },
+            )
+            .post(
+                |axum::extract::State(state),
+                 axum::extract::RawQuery(query),
+                 axum::extract::Json(req)| async move {
+                    axum::response::Json(post(state, None, query, req))
+                },
+            ),
+        )
+        .with_state(State(state));
+    let server = axum::Server::bind(&"0.0.0.0:0".parse().unwrap()).serve(app.into_make_service());
+    let addr = server.local_addr();
+    tokio::spawn(async move { server.await.unwrap() });
+    tokio::time::sleep(tokio::time::Duration::from_millis(1_000)).await;
```

Comment:
I assume you copied this `sleep` from the mocking server of the `legacy` tests. I only had it in for debugging something and forgot to drop it. I think it's safe to not wait here.

Reply:
Fixed in 12d4c68aef74da554a89366f6286cad379e7d870